src/Bicep.Core/Modules/LocalModuleReference.cs

```diff
@@ -49,58 +50,47 @@ public override bool Equals(object? obj)
 
         public static ResultWithDiagnostic<LocalModuleReference> TryParse(string unqualifiedReference, Uri parentModuleUri)
         {
-            if (!Validate(unqualifiedReference, out var failureBuilder))
-            {
-                return new(failureBuilder);
-            }
-
-            return new(new LocalModuleReference(unqualifiedReference, parentModuleUri));
+            return Validate(unqualifiedReference)
+                .Transform(_ => new LocalModuleReference(unqualifiedReference, parentModuleUri));
```

Comment:
this is a really neat pattern I did not know about, TIL

Reply:
I read this article a while back, which was inspiration for the `Result` class (essentially a pattern for error handling without exceptions): https://fsharpforfunandprofit.com/posts/recipe-part2/